nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/client/ESRestClient.java

```diff
@@ -0,0 +1,318 @@
+        HttpEntity entity = new NStringEntity(jsonMapping, ContentType.APPLICATION_JSON);
+        Response response;
+        try {
+            response = lowLevelClient.performRequest("PUT", String.format("/%s/%s/_mapping", indexName, type),
+                    emptyMap(), entity);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+            throw new RuntimeException(String.format("Fail to create mapping on %s/%s: %s", indexName, type, response));
+        }
+
+    }
+
+    @Override
+    public String getNodesInfo() {
+        try {
+            Response response = lowLevelClient.performRequest("GET", "_nodes/_all");
+            return EntityUtils.toString(response.getEntity());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public String getNodesStats() {
+        try {
+            Response response = lowLevelClient.performRequest("GET", "_nodes/stats");
+            return EntityUtils.toString(response.getEntity());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public BulkResponse bulk(BulkRequest request) {
+        try {
+            return client.bulk(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public DeleteResponse delete(DeleteRequest request) {
+        try {
+            return client.delete(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public SearchResponse search(SearchRequest request) {
+        try {
+            return client.search(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public SearchResponse searchScroll(SearchScrollRequest request) {
+        try {
+            return client.searchScroll(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public GetResponse get(GetRequest request) {
+        try {
+            return client.get(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public IndexResponse index(IndexRequest request) {
+        try {
+            return client.index(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public ClearScrollResponse clearScroll(ClearScrollRequest request) {
+        try {
+            return client.clearScroll(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
```

Comment:
I missed that in the interface, but never use `throws Exception`.

Reply:
the interface extends AutoCloseable, code will changed so we don't throw exception in the method